src/System.CommandLine/ParseResult.cs

```diff
@@ -99,7 +100,7 @@ public class ParseResult
         /// Gets the directives found while parsing command line input.
         /// </summary>
         /// <remarks>If <see cref="CommandLineConfiguration.EnableDirectives"/> is set to <see langword="false"/>, then this collection will be empty.</remarks>
-        public IReadOnlyDictionary<string, IReadOnlyList<string>> Directives { get; }
+        public IReadOnlyDictionary<string, IReadOnlyList<string>> Directives => _directives ??= new ();
```

Comment:
in the future we are going to use `IReadOnlyDictionary<TKey, TValue>.Empty` introduced very recently in .NET 8: https://github.com/dotnet/runtime/issues/76028